vpp-manager/hooks/network_manager_hook.go

```diff
@@ -0,0 +1,581 @@
+	}
+
+	// Copy to final location
+	err = os.WriteFile(finalPath, []byte(newContent), 0644)
+	if err != nil {
+		return errors.Wrapf(err, "failed to write final network file")
+	}
+
+	// Create marker file
+	err = os.WriteFile("/host/var/run/vpp/network_file_tweaked", []byte{}, 0644)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create marker file")
+	}
+
+	// Clean up temporary files
+	err = os.Remove(tmpPath)
+	if err != nil && !os.IsNotExist(err) {
+		return errors.Wrapf(err, "failed to remove temporary network file")
+	}
+	err = os.Remove(origPath)
+	if err != nil && !os.IsNotExist(err) {
+		return errors.Wrapf(err, "failed to remove original network file")
+	}
+
+	// Restart systemd-networkd
+	return h.restartService("systemd-networkd")
+}
+
+// removeTweakedNetworkFile removes the tweaked network configuration
+func (h *NetworkManagerHook) removeTweakedNetworkFile() error {
+	markerPath := "/host/var/run/vpp/network_file_tweaked"
+	_, err := os.Stat(markerPath)
+	if os.IsNotExist(err) {
+		return nil
+	}
+
+	h.log.Info("NetworkManagerHook: Deleting tweaked network file...")
+
+	networkFilePath := fmt.Sprintf("/host/etc/systemd/network/%s.network", h.interfaceName)
+	err = os.Remove(networkFilePath)
+	if err != nil && !os.IsNotExist(err) {
+		return errors.Wrapf(err, "failed to remove network file")
+	}
+
+	err = os.Remove(markerPath)
+	if err != nil && !os.IsNotExist(err) {
+		return errors.Wrapf(err, "failed to remove marker file")
+	}
+
+	return nil
+}
+
+// beforeVppRun handles tasks before VPP starts
+func (h *NetworkManagerHook) beforeVppRun() error {
+	// Fix DNS configuration for NetworkManager
+	err := h.fixDNS()
+	if err != nil {
+		return err
+	}
+
+	// Save network file for AWS systemd-networkd
+	err = h.saveNetworkFile()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// vppRunning handles tasks while VPP is running
+func (h *NetworkManagerHook) vppRunning() error {
+	// Restart network services
+	err := h.restartNetwork()
+	if err != nil {
+		return err
+	}
+
+	// Tweak network file for AWS systemd-networkd
+	err = h.tweakNetworkFile()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// vppDoneOk handles cleanup when VPP stops gracefully
+func (h *NetworkManagerHook) vppDoneOk() error {
+	// Remove the DNS fix for NetworkManager
+	err := h.undoDNSFix()
+	if err != nil {
+		return err
+	}
+
+	// Remove the tweaked network file for AWS systemd-networkd
+	err = h.removeTweakedNetworkFile()
+	if err != nil {
+		return err
+	}
+
+	// Restart network services
+	err = h.restartNetwork()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// vppErrored handles cleanup when VPP stops with an error
+func (h *NetworkManagerHook) vppErrored() error {
+	// same cleanup as vppDoneOk()
+	return h.vppDoneOk()
+}
+
+// Execute runs the appropriate hook logic for the given hook point
+func (h *NetworkManagerHook) Execute(hookPoint HookPoint) error {
```

Comment:
Could we add a config option to disable the `NetworkManagerHook` implementation ?
Just in case there is an issue with the implem, or we want to superseed it with script logic